src/Migrator.sol

```diff
@@ -83,44 +83,101 @@ contract Migrator is IMigrator, Initializable, Ownable {
         swapSlippage = _amount;
     }
 
+    /// @inheritdoc IMigrator
+    function setApprovals() public onlyOwner {
+        ERC20(address(weth)).safeApprove(address(balancerVault), type(uint256).max);
+        token.safeApprove(address(balancerVault), type(uint256).max);
+        bpt.safeApprove(address(auraPool), type(uint256).max);
+        ERC20(address(slp)).safeApprove(address(sushiRouter), type(uint256).max);
+    }
+
     /*
-        Public functions
+        External functions
     */
 
     /// @inheritdoc IMigrator
-    function calculateSlpAmounts(uint256 _slpAmount) public view returns (uint256, uint256) {
+    function migrate(bool _stakeBpt) external {
+        uint256 slpBalance = slp.balanceOf(msg.sender);
+
+        ERC20(address(slp)).safeTransferFrom(msg.sender, address(this), slpBalance);
+
+        _unwrapSlp();
+
+        _swapWethForTokenBalancer();
+
+        _depositIntoBalancerPool();
+
+        // msg.sender receives auraBPT or BPT depending on their choice to deposit into Aura pool
+        if (_stakeBpt) _depositIntoRewardsPool();
+        else bpt.safeTransferFrom(address(this), msg.sender, bpt.balanceOf(address(this)));
+
+        emit Migrated(msg.sender, slpBalance, _stakeBpt);
+    }
+
+    /// @inheritdoc IMigrator
+    function userDepositIntoRewardsPool() external {
+        uint256 amount = bpt.balanceOf(msg.sender);
+        bpt.safeTransferFrom(msg.sender, address(this), amount);
+        auraPool.deposit(amount, msg.sender);
+    }
+
+    receive() external payable {
+        if (msg.value > 0) {
```

Comment:
This should have a guard preventing certain groups of people (users, for example) from sending WETH to the contract

Reply:
added guard